src/Summoner/License.hs

```diff
@@ -15,25 +18,47 @@ import qualified Data.Text as T
 -- License
 ----------------------------------------------------------------------------
 
-licenseNames :: [License]
-licenseNames = map fst githubLicenseQueryNames
-
-githubLicenseQueryNames :: [(License, Text)]
-githubLicenseQueryNames =
-    [ ("MIT",        "mit")
-    , ("BSD2",       "bsd-2-clause")
-    , ("BSD3",       "bsd-3-clause")
-    , ("GPL-2",      "gpl-2.0")
-    , ("GPL-3",      "gpl-3.0")
-    , ("LGPL-2.1",   "lgpl-2.1")
-    , ("LGPL-3",     "lgpl-3.0")
-    , ("AGPL-3",     "agpl-3.0")
-    , ("Apache-2.0", "apache-2.0")
-    , ("MPL-2.0",    "mpl-2.0")
-    ]
-
-newtype License = License { unLicense :: Text }
-    deriving (IsString, Eq, Ord, Show)
+data LicenseName = MIT
+             | BSD2
+             | BSD3
+             | GPL2
+             | GPL3
+             | LGPL21
+             | LGPL3
+             | AGPL3
+             | Apache20
+             | MPL20
+             deriving (Eq, Ord, Show, Enum, Bounded, Generic)
+
+githubLicenseQueryNames :: LicenseName -> Text
+githubLicenseQueryNames MIT      = "mit"
+githubLicenseQueryNames BSD2     = "bsd-2-clause"
```

Comment:
This is a nice use-case for `-XLambdaCase` extension! When your function have very long name, you can write like this:

```haskell
githubLicenseQueryNames = \case
    MIT  -> "mit"
    BSD2 -> "bsd-2-clause"
    ...
```